internal/controllers/ingress_controller.go

```diff
@@ -248,6 +260,37 @@ func (irec *IngressReconciler) reconcileDomains(ctx context.Context, ingress *ne
 	return nil
 }
 
+func (irec *IngressReconciler) reconcileTunnels(ctx context.Context, ingress *netv1.Ingress) error {
+	tunnels := ingressToTunnels(ingress)
+
+	for _, tunnel := range tunnels {
+		if err := controllerutil.SetControllerReference(ingress, &tunnel, irec.Scheme); err != nil {
```

Comment:
This isn't a change request, just me clarifying how this works and where we are going with this:

reading through https://book.kubebuilder.io/reference/watching-resources/operator-managed.html?highlight=owner#manage-the-owned-resource later on in their _SetupWithManager_ they add a `.Owns` call for the owned resource. 

I'm assuming that once we get the store up and going and we have it handling tunnels, then that is when the ingress controller will 
- get a reconcile for the ingress
- update it in the store
- calculate intended tunnel differences between the stores ingress and the stores tunnels
- do a kubectl update of the tunnel based on that diff
- receive an update reconcile event for the tunnel *because it is watching/owning that resource*
- update store with new tunnel value

Let me know if thats what you are thinking

Reply:
Yep, that is what we are shooting for!